### PR TITLE
Develop stream 2023 05 18

### PR DESCRIPTION
**Pull request by MKKnorr**
- Use _WIN32 instead of WIN32
- Remove checks for C++11 during compile time
- Fix compilation for `transform_iterator` with `__device__`-only operators
- Remove references to hcc
- Fix some warnings
- Fix missing gtest dependency

### CODE REVIEW

**Comment by MKKnorr** [comment]
Missed a part when rebasing, should be fixed now. The thrust 2.0 update is not included in this PR

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,10 +5,12 @@ Full documentation for rocThrust is available at [https://rocthrust.readthedocs.
 ## (Unreleased) rocThrust 2.18.0 for ROCm 5.6
```
Comment (stanleytsang-amd) [inline CHANGELOG.md:5]: This will go in for ROCm 5.7. I'll update the changelog and the package version in CMake.

Comment (stanleytsang-amd) [reply CHANGELOG.md:5]: Ah, the PR is coming from the Stream repo.  Can you update the changelog entry to 5.7?

Comment (stanleytsang-amd) [reply CHANGELOG.md:5]: @nolmoonen @vince-streamhpc @Snektron Hi, can one of you update the changelog entry to indicate ROCm 5.7? Thanks.  I'd like to get this merged in before Wednesday.

Comment (MKKnorr) [reply CHANGELOG.md]: I changed it. Sorry for the delay

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -60,7 +60,6 @@ if(BUILD_TEST)
       BUILD_PROJECT       TRUE
       UPDATE_DISCONNECTED TRUE
     )
-    list( APPEND CMAKE_PREFIX_PATH ${GTEST_ROOT} )
-    find_package(GTest REQUIRED)
+    find_package(GTest REQUIRED CONFIG PATHS ${GTEST_ROOT})
```
[the remark this answers is not in the document]
Comment (MKKnorr) [reply cmake/Dependencies.cmake:63]: This was necessary due to hipcc not implicitly linking libpthread any more, and it also depends on what cmake version is being used. Never cmake versions don't need this fix

[the remark this answers is not in the document]
Comment (stanleytsang-amd) [reply cmake/Dependencies.cmake:63]: Okay.  Thanks for the explanation.

File: CHANGELOG.md (hunk 1 above)
Comment (stanleytsang-amd) [reply CHANGELOG.md]: No worries. Thanks!

**Review by stanleytsang-amd — APPROVED** [review]

**Pull request merged**